prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -154,17 +167,22 @@ trait Vectors extends Scalars with TimeUnits with Base {
 
     val staleDataLookbackSeconds = 5 * 60 // 5 minutes
 
-    private[prometheus] val (columnFilters, column) = labelMatchesToFilters(mergeNameToLabels)
+    private[prometheus] val (columnFilters, column, bucketOpt) = labelMatchesToFilters(mergeNameToLabels)
 
-    def toPeriodicSeriesPlan(timeParams: TimeRangeParams): PeriodicSeriesPlan = {
+    def toSeriesPlan(timeParams: TimeRangeParams): PeriodicSeriesPlan = {
 
       // we start from 5 minutes earlier that provided start time in order to include last sample for the
       // start timestamp. Prometheus goes back unto 5 minutes to get sample before declaring as stale
-      PeriodicSeries(
+      val ps = PeriodicSeries(
         RawSeries(timeParamToSelector(timeParams, staleDataLookbackSeconds * 1000),
           columnFilters, column.toSeq),
         timeParams.start * 1000, timeParams.step * 1000, timeParams.end * 1000
       )
+      bucketOpt.map { bOpt =>
+        // It's a fixed value, the range params don't matter at all
+        val param = ScalarFixedDoublePlan(bOpt, RangeParams(0, Long.MaxValue, 60000L))
+        ApplyInstantFunction(ps, InstantFunctionId.HistogramBucket, Seq(param))
```

Comment:
Now that I have had a chance to see this PR fully and understand some of the changes that come later on SeriesPlan, I am of the opinion that Histogram Bucket extraction shouldnt be an instant function at the logical plan.

We are including some non-use cases in the logical plan in order to include a corner use case. For example `sum(log(RawSeriesPlan))` (shortened for brevity). Let us discuss offline in more detail.

Reply:
Resolved through offline discussion.... will push changes to make LogicalPlans more type safe.